File: __Viejo/Formularios/Agregar/add_ajuste_stock.vb
Public Class add_ajuste_stock

    Private Sub add_ajuste_stock_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        Me.Dispose()
    End Sub

    Private Sub add_ajuste_stock_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        form = Me

        'Cargo todas los items
        cargar_combo(cmb_items, "SELECT id_item, descript FROM items ORDER BY item ASC", basedb, "descript", "id_item")
        cmb_items.Text = "Seleccione un item..."
    End Sub

    Private Sub cmd_exit_Click(sender As Object, e As EventArgs) Handles cmd_exit.Click
        Me.Dispose()
    End Sub

    Private Sub cmd_ok_Click(sender As Object, e As EventArgs) Handles cmd_ok.Click
        Dim _as As New ajusteStock

        If cmb_items.Text = "Seleccione un item..." Then
            MsgBox("Debe seleccionar un item.", vbExclamation + vbOKOnly, "Centrex")
            Exit Sub
            If txt_cantidad.Text = "" Then
                MsgBox("Debe escribir una cantidad a ajustar.", vbExclamation + vbOKOnly, "Centrex")
                Exit Sub
            End If
        End If

        With _as
            .fecha = Hoy()
            .id_item = cmb_items.SelectedValue
            .cantidad = txt_cantidad.Text
            .notas = txt_notas.Text
        End With

        If Not add_ajusteStock(_as) Then
            MsgBox("Hubo un error al ingresar el ajuste de stock.", vbExclamation + vbOKOnly, "Centrex")
        End If
        closeandupdate(Me)
    End Sub

    Private Sub pic_search_item_Click(sender As Object, e As EventArgs) Handles pic_search_item.DoubleClick
        Dim tmp As String
        tmp = tabla
        'tabla = "items_sinDescuento"
        tabla = "items_registros_stock"
        Me.Enabled = False
        search.ShowDialog()
        tabla = tmp

        'Establezco la opción del combo
        'cmb_items.SelectedIndex = cmb_items.FindString(info_item(id).descript)
        cmb_items.SelectedValue = id
        id = 0
    End Sub

    Private Sub txt_cantidad_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txt_cantidad.KeyPress
        If esNumero(e, True) = 0 Then
            e.KeyChar = ""
        End If
    End Sub
End Class

File: Formularios/Agregar/add_ajuste_stock.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;


namespace Centrex
{
    public partial class add_ajuste_stock
    {
        public add_ajuste_stock()
        {
            InitializeComponent();
        }

        private void add_ajuste_stock_FormClosed(object sender, FormClosedEventArgs e)
        {
            Dispose();
        }

        private void add_ajuste_stock_Load(object sender, EventArgs e)
        {
            // form = Me ' Comentado para evitar error de compilación

            // Cargo todas los items
            Cargar_Combo(
                 ref cmb_items,
                "Items",
                "Descript",
                "IdItem",
                predet: 0,
                soloActivos: true,
                filtros: new Dictionary<string, object> { { "IdProveedor", 1 } },
                orden: new List<Tuple<string, bool>> { Tuple.Create("Descript", true) }
            );
            cmb_items.Text = "Seleccione un item...";
        }

        private void cmd_exit_Click(object sender, EventArgs e)
        {
            Dispose();
        }

        private void cmd_ok_Click(object sender, EventArgs e)
        {
            var _as = new AjusteStockEntity();

            if (cmb_items.Text == "Seleccione un item...")
            {
                Interaction.MsgBox("Debe seleccionar un item.", (MsgBoxStyle)((int)Constants.vbExclamation + (int)Constants.vbOKOnly), "Centrex");
                return;
            }
            else if (string.IsNullOrEmpty(txt_cantidad.Text))
            {
                Interaction.MsgBox("Debe escribir una cantidad a ajustar.", (MsgBoxStyle)((int)Constants.vbExclamation + (int)Constants.vbOKOnly), "Centrex");
                return;
            }

            _as.Fecha = DateOnly.FromDateTime(DateTime.Now);
            _as.IdItem = Conversions.ToInteger(cmb_items.SelectedValue);
            _as.Cantidad = Conversions.ToInteger(txt_cantidad.Text);
            _as.Notas = txt_notas.Text;

            if (!add_ajusteStock(_as))
            {
                Interaction.MsgBox("Hubo un error al ingresar el ajuste de stock.", (MsgBoxStyle)((int)Constants.vbExclamation + (int)Constants.vbOKOnly), "Centrex");
            }
            closeandupdate(this);
        }

        private void pic_search_item_Click(object sender, EventArgs e)
        {
            string tmp;
            tmp = tabla;
            // tabla = "items_sinDescuento"
            tabla = "items_registros_stock";
            Enabled = false;
            My.MyProject.Forms.search.ShowDialog();
            tabla = tmp;

            // Establezco la opción del combo
            // cmb_items.SelectedIndex = cmb_items.FindString(info_item(id).descript)
            cmb_items.SelectedValue = id;
            id = 0;
        }

        private void txt_cantidad_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (Conversions.ToInteger(esNumero(e, true)) == 0)
            {
                e.KeyChar = Conversions.ToChar("");
            }
        }
    }
}
